modules/gapi/src/streaming/onevpl/accelerators/accel_policy_dx11.cpp

```diff
@@ -30,28 +32,59 @@ namespace gapi {
 namespace wip {
 namespace onevpl {
 
-VPLDX11AccelerationPolicy::VPLDX11AccelerationPolicy() :
-    hw_handle(nullptr)
+VPLDX11AccelerationPolicy::VPLDX11AccelerationPolicy(device_selector_ptr_t selector) :
+    VPLAccelerationPolicy(selector),
+    hw_handle(),
+    device_context(),
+    allocator()
 {
-#ifdef CPU_ACCEL_ADAPTER
-    adapter.reset(new VPLCPUAccelerationPolicy);
-#endif
+    // setup dx11 device
+    IDeviceSelector::DeviceScoreTable devices = get_device_selector()->select_devices();
+    GAPI_Assert(devices.size() == 1 && "Multiple(or zero) acceleration devices case is unsupported");
+    AccelType accel_type = devices.begin()->second.get_type();
+    GAPI_Assert(accel_type == AccelType::DX11 &&
+                "Unexpected device AccelType while is waiting AccelType::DX11");
+
+    hw_handle = reinterpret_cast<ID3D11Device*>(devices.begin()->second.get_ptr());
+
+    // setup dx11 context
+    IDeviceSelector::DeviceContexts contexts = get_device_selector()->select_context();
+    GAPI_Assert(contexts.size() == 1 && "Multiple(or zero) acceleration context case is unsupported");
+    accel_type = contexts.begin()->get_type();
+    GAPI_Assert(accel_type == AccelType::DX11 &&
+                "Unexpected context AccelType while is waiting AccelType::DX11");
+    device_context = reinterpret_cast<ID3D11DeviceContext*>(contexts.begin()->get_ptr());
+
+    // setup dx11 allocator
+    memset(&allocator, 0, sizeof(mfxFrameAllocator));
+    allocator.Alloc = alloc_cb;
+    allocator.Lock = lock_cb;
+    allocator.Unlock = unlock_cb;
+    allocator.GetHDL = get_hdl_cb;
+    allocator.Free = free_cb;
+    allocator.pthis = this;
 }
 
 VPLDX11AccelerationPolicy::~VPLDX11AccelerationPolicy()
 {
-    if (hw_handle)
-    {
-        GAPI_LOG_INFO(nullptr, "VPLDX11AccelerationPolicy release ID3D11Device");
-        hw_handle->Release();
+    for (auto& allocation_pair : allocation_table) {
+        allocation_pair.second.reset();
     }
+    GAPI_LOG_INFO(nullptr, "destroyed");
 }
 
 void VPLDX11AccelerationPolicy::init(session_t session) {
-    mfxStatus sts = MFXVideoCORE_GetHandle(session, MFX_HANDLE_D3D11_DEVICE, reinterpret_cast<mfxHDL*>(&hw_handle));
+    mfxStatus sts = MFXVideoCORE_SetHandle(session, MFX_HANDLE_D3D11_DEVICE, (mfxHDL) hw_handle);
```

Comment:
`(mfxHDL)`
Why is it not some_cast<>()?

Reply:
some kind of copy-paste from public VPL examples. I'll fix it later